compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/CompilerDriver.java

```diff
@@ -217,7 +217,15 @@ private boolean stopCompilation(BLangPackage pkgNode, CompilerPhase nextPhase) {
             return true;
         }
         if (pkgNode.containsTestablePkg()) {
-            // We have to check both the compilation unit nodes in the package and testable node
+            // There can be an edge case where the compilation unit nodes in the package node can be empty but the
+            // testable package can have one or many. This has to be treated separately. Check if the package node
+            // is empty and then execute the condition.
+            if (pkgNode.getCompilationUnits().isEmpty()) {
+                return checkNextPhase(nextPhase) &&
+                        (dlog.errorCount > 0 || pkgNode.getTestablePkg().getCompilationUnits().isEmpty());
```

Comment:
Do we have to do the "pkgNode.getTestablePkg().getCompilationUnits().isEmpty()" check? Is it always true inside this if condition?

Reply:
No we don't need it since it'll always be false inside the if condition. Removed the extraneous check with  4dd0fe6